File: backend/app/schemas/permission.py
from typing import List
from uuid import UUID
from pydantic import BaseModel, Field
from app.models.permission import PrincipalType


class PermissionBase(BaseModel):
    principal_type: PrincipalType
    principal_id: str = Field(..., min_length=1)


class PermissionCreate(PermissionBase):
    pass


class PermissionResponse(PermissionBase):
    id: int
    uuid: UUID
    workspace_id: int
    
    model_config = {
        "from_attributes": True
    }


class PermissionBulkCreate(BaseModel):
    permissions: List[PermissionCreate]

File: frontend/src/types/permission.ts
export type PrincipalType = 'USER' | 'GROUP';

export interface Permission {
  id: number;
  uuid: string;
  workspace_id: number;
  principal_type: PrincipalType;
  principal_id: string;
}

export interface PermissionCreate {
  principal_type: PrincipalType;
  principal_id: string;
}